### PR TITLE
Remove run shell from build scripts runner

### PR DESCRIPTION
**Pull request by jelmansouri**
As discussed with @damienmg

### CODE REVIEW

File: cargo/cargo_build_script.bzl [hunk 1]
```diff
@@ -14,6 +14,7 @@ def _cargo_build_script_run(ctx, script):
     compilation_mode_opt_level = get_compilation_mode_opts(ctx, toolchain).opt_level
 
     crate_name = ctx.attr.crate_name
+
```
Comment (damienmg) [inline cargo/cargo_build_script.bzl:17]: nit: unrelated change, remove?

File: cargo/cargo_build_script_runner/bin.rs [hunk 2]
```diff
@@ -53,12 +53,28 @@ fn main() {
...
             let mut command = Command::new(exec_root.join(&progname));
             command
-                .args(args)
                 .current_dir(manifest_dir.clone())
                 .env("OUT_DIR", out_dir_abs)
                 .env("CARGO_MANIFEST_DIR", manifest_dir)
                 .env("RUSTC", rustc);
 
+            while let Some(dep_env_path) = args.next() {
+                if let Ok(contents) = read_to_string(dep_env_path) {
```
Comment (damienmg) [inline cargo/cargo_build_script_runner/bin.rs:62]: opt: add an error handling here? If the file is unreadable, that's probably a worthy failure mode.

File: cargo/cargo_build_script_runner/bin.rs [hunk 3]
```diff
@@ -53,12 +53,28 @@ fn main() {
...
+                    for line in contents.split('\n') {
+                        let mut key_val = line.splitn(2, '=');
+                        match (key_val.next(), key_val.next()) {
+                            (Some(key), Some(value)) => {
+                                command.env(key, value);
+                            }
+                            _ => {
+                                eprintln!("error: Wrong environment file format");
```
Comment (damienmg) [inline cargo/cargo_build_script_runner/bin.rs:70]: This is a "should not happen" error, which means if we fall there we have a serious issue in the build rules, maybe leaves some comment to help the user raise this as a bug?

Comment (jelmansouri) [reply cargo/cargo_build_script_runner/bin.rs:70]: Makes sense
